File: RetrofitTest/app/src/main/java/presenter/FourthPresenter.java
package presenter;

import javax.inject.Inject;

import contract.FourthContract;
import info.LoginNetInfo;

public class FourthPresenter implements FourthContract.Presenter {

    @Inject
    FourthContract.Model model ;

    private FourthContract.View mView ;
    @Inject
    public FourthPresenter(FourthContract.View mView) {
        this.mView = mView;
    }

    @Override
    public void loginNet() {



        model.loginNet(mView.getUserName(), mView.getPassWord(), new LoginNetInfo<String>() {
            @Override
            public void loginNetSuccess(String success) {
                mView.loginSuccess(success);
            }

            @Override
            public void loginNetError(String error) {

                mView.error(error);

            }
        });

    }
}

File: RetrofitTest/app/src/main/java/presenter/FourthPresenter.kt
package presenter

import contract.FourthContract

/**
 * created by shi on 2018/7/6/006
 *
 */
class FourthPresenter constructor(private var mView : FourthContract.View ,private var model : FourthContract.Model<String>) : FourthContract.Presenter {


    override fun loginNet() {

        model.loginNet(mView.getUserName() , mView.getPassWord() , null)


    }
}